src/Security/TokenChecker.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Security;
+
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class TokenChecker
+{
+    /**
+     * @var Session
+     */
+    private $session;
+
+    /**
+     * Constructor.
+     *
+     * @param Session $session
+     */
+    public function __construct(Session $session)
+    {
+        $this->session = $session;
+    }
+
+    /**
+     * Checks if an authenticated token exists in the session.
+     *
+     * @param string $sessionKey
+     *
+     * @return bool
+     */
+    public function isAuthenticated(string $sessionKey): bool
+    {
+        return null !== $this->getToken($sessionKey);
```

Comment:
Why don't you use the `isAuthenticated` method on the token?

Reply:
The `getToken` method already does: https://github.com/contao/core-bundle/pull/1232/files/df31357d729b02ffd8d54eac331a22c29efbb83e#diff-c81a5fe77c0d656a71ce6b60c6bf87b5R80

Reply:
Ah, missed that!